### PR TITLE
Fix #689: Do not log ACTIVATION_VERSION_CHANGED when preparing the activation

### LINKED ISSUE #689: Do not log ACTIVATION_VERSION_CHANGED when preparing the activation
Currently, we incorrectly log activation status change reason whenever the activation goes to PENDING_COMMIT:

Screenshot 2022-05-09 at 14 53 16

https://github.com/wultra/powerauth-server/blame/32df0accfe80f46581ea9936cd818747157c5dea/powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/v3/ActivationServiceBehavior.java#L948

### PR DESCRIPTION
**Pull request by petrdvorak**

### CODE REVIEW

File: powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/database/model/AdditionalInformation.java [hunk 1]
```diff
@@ -24,24 +24,63 @@
...
-     * <li>during upgrade commit (version changes to a newer one)</li>
-     * </ul>
-     */
-    public static final String ACTIVATION_VERSION_CHANGED = "ACTIVATION_VERSION_CHANGED";
+    public static class Key {
+        
+        /**
+         * Key describing reasons for blocked activation.
+         */
+        public static final String BLOCKED_REASON = "BLOCKED_REASON";
```
Comment (banterCZ) [inline powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/database/model/AdditionalInformation.java:32]: When refactoring, can we change it to enum (maybe in s separate PR)?

File: powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/v3/ActivationServiceBehavior.java [hunk 2]
```diff
@@ -945,7 +945,7 @@ public PrepareActivationResponse prepareActivation(String activationCode, String
             String mac = BaseEncoding.base64().encode(responseCryptogram.getMac());
 
             // Persist activation report and notify listeners
-            activationHistoryServiceBehavior.saveActivationAndLogChange(activation, null, AdditionalInformation.ACTIVATION_VERSION_CHANGED);
```
Comment (banterCZ) [inline powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/v3/ActivationServiceBehavior.java:948]: Ok, so it seems that the most important change is this one.

**Review by banterCZ — APPROVED** [review]
Seems ok.

File: powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/v3/ActivationServiceBehavior.java (hunk 2 above)
Comment (banterCZ) [reply powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/v3/ActivationServiceBehavior.java:948]: Just reminding that in the original issue #501 there was:

> during activation prepare (null version -> specific version)

File: powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/database/model/AdditionalInformation.java (hunk 1 above)
Comment (petrdvorak) [reply powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/database/model/AdditionalInformation.java:32]: Let's do this in a separate issue. The benefit of the change is minimal.

**Pull request merged**